orderbook/src/database/orders.rs

```diff
@@ -77,13 +88,28 @@ impl Database {
                 COALESCE(SUM(t.buy_amount), 0) AS sum_buy, \
                 COALESCE(SUM(t.fee_amount), 0) AS sum_fee \
             FROM orders o LEFT OUTER JOIN trades t ON o.uid = t.order_uid
-            WHERE o.valid_to >= $1 \
+            WHERE \
+                o.valid_to >= $1 AND \
+                ($2 OR o.owner = $3) AND \
+                ($4 OR o.sell_token = $5) AND \
+                ($6 OR o.buy_token = $7) \
             GROUP BY o.uid;";
         // To get only not fully executed orders we probably want to use something like
         // `HAVING sum_sell < orders.sell_amount` but still need to find the best way to make this
         // pick the correct column based on order type.
+
+        fn as_bytes_or_default(h160: Option<&H160>) -> &[u8] {
+            h160.map(|h160| h160.as_bytes()).unwrap_or_default()
+        }
+
         sqlx::query_as(QUERY)
-            .bind(min_valid_to)
+            .bind(filter.min_valid_to)
+            .bind(filter.owner.is_none())
```

Comment:
nit: I think you should just be able to pass in the option directly and check e.g. for `$3 is NULL OR o.owner = $3` (https://docs.rs/sqlx/0.4.2/sqlx/macro.query.html#nullability-bind-parameters)

Reply:
Nice suggestion, implemented